controllers/istiorevision/istiorevision_controller.go

```diff
@@ -64,17 +65,19 @@ const istioCniName = "default"
 // Reconciler reconciles an IstioRevision object
 type Reconciler struct {
 	client.Client
-	Config       config.ReconcilerConfig
-	Scheme       *runtime.Scheme
-	ChartManager *helm.ChartManager
+	Config             config.ReconcilerConfig
+	Scheme             *runtime.Scheme
+	ChartManager       *helm.ChartManager
+	cniDependencyCache sync.Map
```

Comment:
Why is a cache needed here?

Reply:
Since the `DependsOnIstioCNI` method is invoked as part of `Reconcile` method, I want to avoid invoking the `ComputeValues` method unless there is any change in the IstioRevision.

Reply:
OK but does `ComputeValues` take a long time or access any resources on the cluster?

AFAICT it just computes stuff in memory, which should be relatively fast..

The reason I ask is because caches introduce complexity, so unless they save us substantial time/resources then I think we best avoid them.

Also if you believe it's necessary I'd document the reason why it's needed.

Reply:
Right, on further testing, I found one edge case with Cache when users create, delete and re-creates the same resource. Let me remove the cache.